### PR TITLE
Add instructions for `CodeCellPlaceholder` component

### LINKED ISSUE #2173: Document how to use CodeCellPlaceholder
## Problem

`@kaelynj` needs to add a non-trivial amount of setup code for an upcoming Jupyter notebook guide. Our docs infrastructure requires that all code be self-contained in a single Jupyter notebook, or to create and publish a third-party library with the helper code. So, this means that the setup code needs to live in the guide.

The code needs to live in a `code` block, not a `markdown` block, because the variables and functions defined need to exist for later code blocks in the file.

We don't want to show all the setup code by default to the user because it's too verbose. So, we should instead have a way to collapse `code` blocks into `<details>`, aka "a twisty" like this:

Image

The code itself _must_ be a code block and not a markdown block. We need some way to tell the docs renderer in closed source to put this code into a `<details>`. 

## Solution already exists! CodeCellPlaceholder

This mechanism allows you to define code anywhere in your notebook, and then have the docs renderer dynamically move the code to render somewhere else. For example, you can have this structure

- code cell with all the setup code. You should set a tag like `id-setup-code`
- markdown cell with

```mdx
<details><summary>description</summary>
   <CodeCellPlaceholder tag="id-setup-code" />
</details>
```

## This issue: document how it works

### PR DESCRIPTION
**Pull request by frankharkins**
Closes #2173. Also documents the `CodeAssistantAdmonition` component.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -663,6 +676,36 @@ There is a specific use case where you want to show instructions for different o
...
+### CodeCellPlaceholder
+
+This component only works in notebooks. Notebook code cells are always at the
+top-level of content, but sometimes you'll want to have them nested in other
+components, such as in tabs or in a list. To use this component, add a tag
+starting with `id-` to the cell you'd like to move, then add a
+`<CodeCellPlaceholder tag="id-tag" />` component with the same tag somewhere in
+your markdown. This will move that code cell into the place of the component.
+
+You can then use this component anywhere in your mdx. While you can move code
```
Comment (Eric-Arellano) [inline README.md:688]: MDX is probably confusing

File: README.md [hunk 2]
```diff
@@ -663,6 +676,36 @@ There is a specific use case where you want to show instructions for different o
...
+cells anywhere, try to keep them relatively close to their position in the
+notebook and preserve their order to avoid confusion.
+
+```mdx
+Here's a list where the second list element is a code cell.
+The cell with tag "id-move-to-list" will be moved inside this list.
```
Comment (Eric-Arellano) [inline README.md:694]: It would probably be more clear to show an actual Jupyter notebook, like the JSON for it, with the two code blocks

**Review by Eric-Arellano — COMMENTED** [review]
Thank you!

File: README.md [hunk 3]
```diff
@@ -592,6 +592,19 @@ By default, the title is the `type` capitalized. You can customize it by setting
...
 
+We also have a specialized admonition for Qiskit Code Assistant prompt suggestions.
+
+```mdx
+<CodeAssistantAdmonition
+  tagLine="Need help? Try asking Qiskit Code Assistant."
+  prompts={[
+    "# Print the version of Qiskit we're using",
+    "# Return True if the version of Qiskit is 1.0 or greater",
+    "# Install Qiskit 1.0.2",
```
[the remark this answers is not in the document]
Comment (frankharkins) [reply README.md:603]: Hahaha 😭

File: README.md [hunk 4]
```diff
@@ -592,6 +592,19 @@ By default, the title is the `type` capitalized. You can customize it by setting
 </Admonition>
 ```
 
+We also have a specialized admonition for Qiskit Code Assistant prompt suggestions.
```
[the remark this answers is not in the document]
Comment (frankharkins) [reply README.md:595]: Good call, hopefully this won't drive people too crazy

File: README.md (hunk 1 above)
Comment (frankharkins) [reply README.md:694]:
```suggestion
You can then use this component anywhere in your markdown. While you can move code
```

File: README.md [hunk 5]
```diff
@@ -663,6 +676,65 @@ There is a specific use case where you want to show instructions for different o
...
+This component only works in notebooks. Notebook code cells are always at the
+top-level of content, but sometimes you'll want to have them nested in other
+components, such as in tabs or in a list. While you could write your code
+as a markdown block, it's usually preferable to keep the code as a code block
+so that it is executed and its code can be later used in the notebook. The
+CodeCellPlaceholder component allows you to still use a code block, but move
+it to render somewhere else in the notebook.
+
+To use this component, add a tag
+starting with `id-` to the cell you'd like to move, then add a
```
Comment (Eric-Arellano) [inline README.md:690]:
```suggestion
starting with `id-` to the code cell you'd like to move, then add a
```

File: README.md [hunk 6]
```diff
@@ -663,6 +676,65 @@ There is a specific use case where you want to show instructions for different o
...
+`<CodeCellPlaceholder tag="id-tag" />` component with the same tag somewhere in
+your markdown. This will move that code cell into the place of the component.
+
+You can then use this component anywhere in your markdown. While you can move code
+cells anywhere, try to keep them relatively close to their position in the
+notebook and preserve their order to avoid confusion.
+
+Here's an example of what this might look like in your notebook source.
```
Comment (Eric-Arellano) [inline README.md:698]: Nice example! It may be more clear to put the code cell above the markdown cell. That better mirrors how you wrote the instructions

**Review by Eric-Arellano — APPROVED** [review]

File: .prettierignore [hunk 7]
```diff
@@ -3,3 +3,4 @@ translations
 scripts/js/lib/api/testdata
 .mypy_cache
 public/images
+README.md
```
Comment (frankharkins) [inline .prettierignore:6]: Prettier wanted the MDX example in the README to have a trailing comma, which wouldn't be valid JSON.

````diff
--- a/README.md
+++ b/README.md
@@ -600,7 +600,7 @@ We also have a specialized admonition for Qiskit Code Assistant prompt suggestio
   prompts={[
     "# Print the version of Qiskit we're using",
     "# Return True if the version of Qiskit is 1.0 or greater",
-    "# Install Qiskit 1.0.2"
+    "# Install Qiskit 1.0.2",
   ]}
 />
 ```
````

I've just ignored this; it doesn't seem to be a problem with other MDX files, and I don't think we need to lint the README.

Comment (Eric-Arellano) [reply .prettierignore:6]: Sg. Formatting the README has caught contributors in the past in an annoying way

**Pull request merged**